### DISCUSSION TITLE
How to generate delta for new files only?

### DISCUSSION THREAD

**Discussion by sjurgis** [scolladon/sfdx-git-delta#719] [Q&A] [answered]
We've partly implemented apex pmd (via sfdx scanner) by cleaning up some of the classes that don't have too many problems. However, we have a ton of classes that with high cyclomatic complexity, etc that is unfeasible to solve immediately.
The goal is to prevent adding new files to repo with high complexity.  The way I run scanner right now is `--generate-delta` flag that creates a folder with modified files.
Would it be possible to implement similar flag that creates folder only with newly added files?

**Comment by scolladon** [comment]
Hi @sjurgis,

`--generate-delta` generate the incremental content with both added and modified metadata.
What you could do to have both added and modified changes separated into different commit.
Then you can run the plugin twice to compare those commits with its previous one and output that in a different folder.

Example:
```sh
# create new classes in the repository
...

# create a commit to store those classes
git commit -am 'feat: add new classes'

# modify existing classes in the repository
...

# create a commit to store the modification on those classes
git commit -am 'feat: modify classes'

# create folders for the incremental packages
mkdir added
mkdir modified

# run SGD for added only
sfdx sgd:source:delta -f "HEAD~2" -t "HEAD~1" -d -o added

# run SGD for modified only
sfdx sgd:source:delta -f "HEAD~1" -t "HEAD" -d -o modified

**Comment by sjurgis** [comment, marked as answer]
Hmmm, thanks for posting this. I googled a bit and turns out solution might be much simpler - I can just [grab a list added files](https://stackoverflow.com/questions/61215061/how-to-find-added-files-between-2-commits#comment108296041_61215670) and pass those to sfdx scanner.
i.e.
```
    git fetch origin $BITBUCKET_PR_DESTINATION_BRANCH
    FROM=$(git merge-base $BITBUCKET_COMMIT origin/$BITBUCKET_PR_DESTINATION_BRANCH)
    echo 'Common ancestor: '$FROM
    git diff --name-only --diff-filter=A $FROM..$BITBUCKET_COMMIT
    sfdx scanner:run --target $(git diff --name-only --diff-filter=A $FROM..$BITBUCKET_COMMIT)
```

**Reply by scolladon** [reply]
Very smart !

Now you show it, I think I misunderstood your question at first 😄
Sorry if I misled you.

Thanks for posting here the solution you found.
